prbt_support/test/acceptance_tests/acceptance_test_joint_limits.py

```diff
@@ -0,0 +1,237 @@
+        return success
+
+    def _joint_limit_reaching_test(self, joint_name):
+        """ Test if the robot can be commanded to move exactly to the limits
+
+            Test Sequence:
+              1. Command a movement to the home position.
+              2. Command a movement to the lower limit.
+              3. Command a movement to the upper limit.
+
+            Expected Results:
+              1. Trajectory is executed successfully.
+              2. Trajectory is executed successfully.
+              3. Trajectory is executed successfully.
+        """
+        self._drive_home()
+
+        index = self.joint_names.index(joint_name)
+        limit = _JOINT_LIMITS_DEGREE[joint_name]
+
+        lower_positions = [0] * len(self.joint_names)
+        lower_positions[index] = -radians(limit)
+
+        self.assertTrue(self._execute_trajectory(lower_positions))
+
+        upper_positions = [0] * len(self.joint_names)
+        upper_positions[index] = radians(limit)
+
+        self.assertTrue(self._execute_trajectory(upper_positions))
+
+    def _joint_limit_overstepping_test(self, joint_name):
+        """ Test if the robot does not overstep the limits
+
+            Test Sequence:
+              1. Command a movement to the home position.
+              2. Command a movement overstepping the lower limit.
+              3. Command a movement overstepping the upper limit.
+
+            Expected Results:
+              1. Trajectory is executed successfully.
+              2. Trajectory execution is aborted and the robot does not overstep the limits.
+              3. Trajectory execution is aborted and the robot does not overstep the limits.
+        """
+        self._drive_home()
+
+        index = self.joint_names.index(joint_name)
+        limit = _JOINT_LIMITS_DEGREE[joint_name]
+
+        lower_positions = [0] * len(self.joint_names)
+        lower_positions[index] = -(radians(limit) + _JOINT_LIMIT_OVERSTEP)
+
+        self.assertFalse(self._execute_trajectory(lower_positions))
+        self._check_joint_limits()
+
+        upper_positions = [0] * len(self.joint_names)
+        upper_positions[index] = radians(limit) + _JOINT_LIMIT_OVERSTEP
+
+        self.assertFalse(self._execute_trajectory(upper_positions))
+        self._check_joint_limits()
+
+    def test_joint_limits_reaching(self):
+        """ Perform all reaching tests. Before each test ask the user if he wants to skip it.
+        """
+        for name in self.joint_names:
+            if self._ask_for_permission('joint_limit_reaching_test for ' + name):
+                self._joint_limit_reaching_test(name)
+
+    def test_joint_limits_overstepping(self):
+        """ Perform all overstepping tests. Before each test ask the user if he wants to skip it.
+        """
+        for name in self.joint_names:
+            if self._ask_for_permission('joint_limit_overstepping_test for ' + name):
+                self._joint_limit_overstepping_test(name)
```

Comment:
_Documenting what I tested:_
- Joint 4: successful
- Joint 5: not (because of gripper)
- Joint 6: successful
- Joint 1: successful
- Joint 2: not (potential collision with table)
- Joint 3: successful

(in that order, what was a little unexpected)

Reply:
These are the expected result.